docs/reference/aggregations/pipeline/movfn-aggregation.asciidoc

```diff
@@ -0,0 +1,575 @@
+[[search-aggregations-pipeline-movfn-aggregation]]
+=== Moving Function Aggregation
+
+Given an ordered series of data, the Moving Function aggregation will slide a window across the data and allow the user to specify a custom
+script that is executed on each window of data.  For convenience, a number of common functions are predefined such as min/max, moving averages,
+etc.
+
+This is conceptually very similar to the <<search-aggregations-pipeline-movavg-aggregation, Moving Average>> pipeline aggregation, except
+it provides more functionality and better flexibility.
+==== Syntax
+
+A `moving_fn` aggregation looks like this in isolation:
+
+[source,js]
+--------------------------------------------------
+{
+    "moving_fn": {
+        "buckets_path": "the_sum",
+        "window": 10,
+        "script": "MovingFunctions.windowMin(values)"
+    }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+.`moving_avg` Parameters
+|===
+|Parameter Name |Description |Required |Default Value
+|`buckets_path` |Path to the metric of interest (see <<buckets-path-syntax, `buckets_path` Syntax>> for more details |Required |
+|`window` |The size of window to "slide" across the histogram. |Optional |`5`
+|`script` |The script that should be executed on each window of data |Required |
+|`params` |User-defined extra parameters to be injected into the script |Optional |
+|===
+
+`moving_fn` aggregations must be embedded inside of a `histogram` or `date_histogram` aggregation.  They can be
+embedded like any other metric aggregation:
+
+[source,js]
+--------------------------------------------------
+POST /_search
+{
+    "size": 0,
+    "aggs": {
+        "my_date_histo":{                <1>
+            "date_histogram":{
+                "field":"date",
+                "interval":"1M"
+            },
+            "aggs":{
+                "the_sum":{
+                    "sum":{ "field": "price" } <2>
+                },
+                "the_movavg": {
+                    "moving_fn": {
+                        "buckets_path": "the_sum", <3>
+                        "window": 10,
+                        "script": "MovingFunctions.simpleMovAvg(values)"
+                    }
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:sales]
+
+<1> A `date_histogram` named "my_date_histo" is constructed on the "timestamp" field, with one-day intervals
+<2> A `sum` metric is used to calculate the sum of a field.  This could be any metric (sum, min, max, etc)
```

Comment:
I think we should say `this could be any numeric metric as I don't think you could point a `moving_fn` at, for example, a geo_centroid agg?